core/FusionDecode.cpp

```diff
@@ -0,0 +1,220 @@
+// TODO: add Condor header - this is a new file
+// contact jeff at condor
+#include "FusionTypes.hpp"
+#include "Decode.hpp"
+
+#include "sparta/events/StartupEvent.hpp"
+#include "sparta/utils/LogUtils.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <set>
+#include <map>
+#include <vector>
+
+using namespace std;
+using namespace fusion;
+
+namespace olympia
+{
+// -------------------------------------------------------------------
+// Remove the ghost fusion ops
+// Kept, but currently not called
+// -------------------------------------------------------------------
+void Decode::whereIsEgon(InstGroupPtr &insts,uint32_t &numGhosts)
+{
+    if(insts == nullptr) {
+        //DLOG("whereIsEgon() called with insts null");
+        return;
+    }
+
+    //DLOG("BEFORE # --------------------------------------");
+    //infoInsts(cout,insts);
+
+    for(auto it = insts->begin(); it != insts->end(); ) {
+        if ((*it)->getExtendedStatus()
+                   == olympia::Inst::Status::FUSION_GHOST)
+        {
+            ++numGhosts;
+            it = insts->erase(it);
+        } else {
+            ++it;
+        }
+    }
+    //DLOG("AFTER  # --------------------------------------");
+    //infoInsts(cout,insts);
+    //DLOG("DONE   # --------------------------------------");
+}
+// -------------------------------------------------------------------
+// -------------------------------------------------------------------
+void Decode::processMatches(MatchInfoListType &matches,
+                            InstGroupPtr &insts,
+                            const InstUidListType &inputUids)
+{
+    if(matches.size() == 0) return;
+
+    //Kept for upcoming additional work on reducing ROB entry requirements
+    //DLOG("PROCESS MATCHES "<<matches.size());
+    //DLOG("FUSION: REF   UIDS:");
+    //fuser->infoUids(cout,inputUids,"before "); cout<<endl;
+    //fuser->info(cout,matches,inputUids); cout<<endl;
+    //size_t idx = 0;
+
+    for(const auto &m : matches)
```

Comment:
Suggest another variable name other than `m`.  If I wanted to search for the use of `m`, I get a LOT of hits.

Reply:
i fixed this one, the q/a/b/m.